### ISSUE TITLE
How to dynamically import packages?

### ISSUE THREAD

**Issue by rusmux** [omni-us/jsonargparse#335]
I'm designing a package that have main functionality and optional functionality that requires extra packages to be installed. I want to build a CLI for the package. `jsonargparse` dissuades use of `parse_known_args()`, so I cannot import packages only when needed. 

For example, I want to add optional function arguments. I cannot use `parser.add_function_args` because it would need me to import the function. I must add arguments manually, but I can't always do this, because some arguments may depend on classes in my optional functionality. How to manage this case? Thank you

**Comment by mauvilsa** [comment]
> I cannot use `parser.add_function_args` because it would need me to import the function. 

You could do like:

```python
if optional_package_available:
    import optional_package
...
if optional_package_available:
    parser.add_function_args(optional_package.some_function, ...)
```

> I can't always do this, because some arguments may depend on classes in my optional functionality

Not really sure what you mean by this.

**Comment by rusmux** [comment]
Yes, but if I try to run a command with arguments, it won't recognize them and exit, and the only way to show a message about installing additional packages, as I understand, is to catch `SystemExit`, which I think is not the most elegant solution.

**Comment by mauvilsa** [comment]
I guess what you need is an `Action` that always raises an argument exception with some message. But this action must be triggered if nested arguments are needed. That is, something like:

```python
if optional_package_available:
    from optional_package import some_function
...
if optional_package_available:
    parser.add_function_arguments(some_function, "group")
else:
    parser.add_argument("--group", action=ActionNotAvaliable("please install optional_package"))
```

The CLI would exit with the given message if any option `--group.*` is given. This action would be useful for the case of optional packages, but also for deprecated options that were removed.

`jsonargparse` internally importing optional packages, I am not sure it is a good idea.

**Comment by mauvilsa** [comment]
@rusmux even though it is simple, took me a while to get to this, but now I have a pull request. Please have a look at https://github.com/omni-us/jsonargparse/pull/759/files. Hopefully it would still be useful.
